codenav/default_eval_spec.py

```diff
@@ -0,0 +1,146 @@
+            repo_description=episode_kwargs["repo_description"]
+        )
+        prompt = prompt_builder.build(
+            dict(
+                AVAILABLE_ACTIONS=episode_kwargs["allowed_actions"],
+                RESERVED_CHARACTERS=RESERVED_CHARACTERS,
+                RETRIEVALS_PER_KEYWORD=episode_kwargs["retrievals_per_keyword"],
+            )
+        )
+
+        return Episode(
+            agent=OpenAICodeNavAgent(
+                prompt=prompt,
+                model=episode_kwargs["llm"],
+                allowed_action_types=episode_kwargs["allowed_actions"],
+            ),
+            action_type_to_env=dict(
+                code=PythonCodeEnv(
+                    code_dir=episode_kwargs["code_dir"],
+                    sys_paths=episode_kwargs["sys_paths"],
+                    working_dir=episode_kwargs["working_dir"],
+                ),
+                search=RetrievalEnv(
+                    code_retriever=EsCodeRetriever(
+                        index_name=episode_kwargs["index_name"],
+                        host=episode_kwargs["host"],
+                    ),
+                    expansions_per_query=episode_kwargs["retrievals_per_keyword"],
+                    prototypes_per_query=episode_kwargs["prototypes_per_keyword"],
+                ),
+                done=DoneEnv(),
+            ),
+            user_query_str=eval_input.query,
+        )
+
+    @staticmethod
+    def run_interaction(
+        episode: Episode,
+        interaction_kwargs: Optional[Str2AnyDict] = None,
+    ) -> Str2AnyDict:
+        assert interaction_kwargs is not None
+        episode.step_until_max_steps_or_success(
+            max_steps=interaction_kwargs["max_steps"],
+            verbose=interaction_kwargs["verbose"],
+        )
+        ipynb_str = episode.to_notebook(cur_dir=episode.code_env.working_dir)
+        return dict(ipynb_str=ipynb_str)
+
+    @staticmethod
+    def log_output(
+        interaction_output: Str2AnyDict,
+        eval_input: EvalInput,
+        logging_kwargs: Optional[Str2AnyDict] = None,
+    ) -> Any:
+        assert logging_kwargs is not None
+
+        outfile = os.path.join(logging_kwargs["out_dir"], f"{eval_input.uid}.ipynb")
+        with open(outfile, "w") as f:
+            f.write(interaction_output["ipynb_str"])
+
+        return outfile
+
+
+def run_codenav_on_query(
+    exp_name: str,
+    out_dir: str,
+    query: str,
+    code_dir: str,
+    index_name: str,
+    working_dir: str = os.path.join(
+        os.path.dirname(ABS_PATH_OF_CODENAV_DIR), "playground"
+    ),
+    sys_paths: Optional[List[str]] = None,
+    repo_description_path: Optional[str] = None,
+    es_host: str = DEFAULT_ES_HOST,
+    max_steps: int = 20,
+):
+
+    prompt_dirs = [PROMPTS_DIR]
+    repo_description = "default/repo_description.txt"
+    if repo_description_path is not None:
+        prompt_dir, repo_description = os.path.split(repo_description_path)
+        prompt_dirs.append(prompt_dir)
```

Comment:
I don't love this way of handling the repo description as this means that the code will first search for a file with name
```python
os.path.join(PROMPTS_DIR, repo_description)
```
and, if it finds it, will use that. This should be rare but would be unexpected behavior. You could change it to be
```python
prompt_dirs.insert(0, prompt_dir)
```
but this has its own problems as we'll possibly be using other prompts from `prompt_dir` rather than those from codenav. 

Perhaps we can just add an:
```python
assert not os.path.exists(os.path.join(PROMPTS_DIR, repo_description)), f"{os.path.exists(os.path.join(PROMPTS_DIR, repo_description))} should not exist when attempting to use `prompt_dir={repo_description_path}`"
```?

Reply:
good catch! 

How about this:
```python
prompt_dirs = [PROMPTS_DIR]
    repo_description = "default/repo_description.txt"
    if repo_description_path is not None:
        prompt_dir, repo_description = os.path.split(repo_description_path)
        conflict_path = os.path.join(PROMPTS_DIR, repo_description)
        if os.path.exists(conflict_path):
            raise ValueError(
                f"Prompt conflict detected: {repo_description} already exists in {PROMPTS_DIR}. "
                f"Please rename the {repo_description_path} file to resolve this conflict."
            )
        prompt_dirs.append(prompt_dir)
```

Reply:
LGTM :)!